### PR TITLE
Create a session handle once in the transport ctor and reuse it for all requests rather than creating a new one each time.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1098
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/3548

Verified the performance regression observed due to the change https://github.com/Azure/azure-sdk-for-cpp/pull/3366 from has been resolved.
Now, the TCP connection is being re-used rather than new one being opened for each request.

cc @mikeharder

### CODE REVIEW

File: sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp [hunk 1]
```diff
@@ -253,33 +253,30 @@ void WinHttpTransport::CreateSessionHandle(std::unique_ptr<_detail::HandleManage
...
   {
     GetErrorAndThrow("Error while enforcing TLS 1.2 for connection request.");
   }
 }
 
+WinHttpTransport::WinHttpTransport(WinHttpTransportOptions const& options)
+    : m_options(options), m_sessionHandle(NULL)
+{
+  CreateSessionHandle();
+}
```
Comment (vhvb1989) [inline sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp:278]:
```suggestion
WinHttpTransport::WinHttpTransport(WinHttpTransportOptions const& options)
    : m_options(options), m_sessionHandle(CreateSessionHandle())
{
}
```

File: sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp [hunk 2]
```diff
@@ -141,8 +133,9 @@ namespace Azure { namespace Core { namespace Http {
   class WinHttpTransport final : public HttpTransport {
   private:
     WinHttpTransportOptions m_options;
+    HINTERNET m_sessionHandle;
```
Comment (vhvb1989) [inline sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp:136]:
```suggestion
    HINTERNET m_sessionHandle = NULL;
```

**Review by vhvb1989 — APPROVED** [review]
LGTM

Adding a tiny suggestion

**Comment by vhvb1989** [comment]
Add changelog as well, before merging

**Comment by ahsonkhan** [comment]
Before March (i.e. excluding [this](https://github.com/Azure/azure-sdk-for-cpp/pull/3366) change from main):
Completed 1,103 operations in a weighted-average of 5s (219.756618 ops/s, 0.00455049 s/op)

After March (with [this](https://github.com/Azure/azure-sdk-for-cpp/pull/3366) change, i.e. main currently):
> Completed 291 operations in a weighted-average of 5s (58.126332 ops/s, 0.0172039 s/op)

Now (after [this](https://github.com/Azure/azure-sdk-for-cpp/pull/3585) fix):
> Completed 1,386 operations in a weighted-average of 5s (276.825455 ops/s, 0.00361238 s/op)

From a sanity check, it looks like we improved perf by ~25% since the March release with this fix. @mikeharder It would be worth validating the improvement with more rigorous testing and comparison.

**Comment by mikeharder** [comment]
@ahsonkhan: I can test this by EOD tomorrow.

**Pull request merged**